paddle/fluid/framework/ir/layer_norm_fuse_pass.cc

```diff
@@ -0,0 +1,168 @@
+
+    const auto& gamma_shape = gamma->Var()->GetShape();
+    const auto& beta_shape = beta->Var()->GetShape();
+    const auto& x_shape = x->Var()->GetShape();
+    int64_t x_last_dim = x_shape.back();
+
+    PADDLE_ENFORCE_EQ(gamma_shape.size(), 1,
+                      platform::errors::InvalidArgument(
+                          "The LayerNorm gamma "
+                          "(scale) tensor shape must be one-dimensional, "
+                          "but is %s.",
+                          gamma_shape.size()));
+    PADDLE_ENFORCE_EQ(beta_shape.size(), 1,
+                      platform::errors::InvalidArgument(
+                          "The LayerNorm beta "
+                          "(shift) tensor shape must be one-dimensional, "
+                          "but is %s.",
+                          beta_shape.size()));
+    PADDLE_ENFORCE_EQ(beta_shape, gamma_shape,
+                      platform::errors::InvalidArgument(
+                          "The LayerNorm beta "
+                          "and gamma tensors shapes' must be equal."));
+    PADDLE_ENFORCE_EQ(gamma_shape.front(), x_last_dim,
+                      platform::errors::InvalidArgument(
+                          "The LayerNorm beta "
+                          "and gamma tensors shapes' must be equal to the last "
+                          "input's dimension size."));
+
+    // ------------------ op creation and placement ---------------------------
+
+    OpDesc ln_op_desc;
+    ln_op_desc.SetType("layer_norm");
+    ln_op_desc.SetInput("X", {x->Name()});
+    ln_op_desc.SetInput("Scale", {gamma->Name()});
+    ln_op_desc.SetInput("Bias", {beta->Name()});
+    ln_op_desc.SetOutput("Y", {shift_out->Name()});
+    ln_op_desc.SetAttr("begin_norm_axis", static_cast<int>(x_shape.size() - 1));
+    ln_op_desc.SetAttr("epsilon", *(eps_tensor->data<float>()));
+    ln_op_desc.SetAttr("is_test", true);
+    Node* ln_op = g->CreateOpNode(&ln_op_desc);
+
+    IR_NODE_LINK_TO(x, ln_op);
+    IR_NODE_LINK_TO(gamma, ln_op);
+    IR_NODE_LINK_TO(beta, ln_op);
+    IR_OP_VAR_LINK(ln_op, shift_out);
+    GraphSafeRemoveNodes(
+        g,
+        {x_mean, x_mean_out, x_sub_mean, x_sub_mean_out, sqr_pow,
+         x_sub_mean_sqr, x_sub_mean_sqr_out, std_dev, std_dev_out, eps,
+         std_dev_eps, std_dev_eps_out, std_dev_eps_sqrt, std_dev_eps_sqrt_out,
+         division, division_out, scale, scale_out, shift});
+    found_layer_norm_count++;
+  };
+
+  gpd(graph, handler);
+  AddStatis(found_layer_norm_count);
+  PrettyLogDetail("---    Fused %d subgraphs into layer_norm op.",
+                  found_layer_norm_count);
+}
+
+}  // namespace ir
+}  // namespace framework
+}  // namespace paddle
+
+REGISTER_PASS(layer_norm_fuse_pass, paddle::framework::ir::LayerNormFusePass);
+REGISTER_PASS_CAPABILITY(layer_norm_fuse_pass)
+    .AddCombination(
+        paddle::framework::compatible::OpVersionComparatorCombination()
+            .EQ("elementwise_add", 0)
```

Comment:
.EQ("elementwise_add", 1)  ?

Reply:
I've corrected the `EQ` condition to `GE` - now it works. Nor do I can find the kernel for this op :) However it is somehow automatically generated: https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/fluid/layers/ops.py#L40

Reply:
If keep  .GE("elementwise_add", 0), for any update of elementwise_add, elementwise_sub op etc, layer_norm_fuse will not detect the changes. What if some attribute name changed etc...

Reply:
@lidanqing-intel I've updated version conditions since I've added validating `reduce_mean` attributes. Please take a look.